### PR TITLE
MCLOUD-5818: Fix for composer running out of memory in container

### PR DESCRIPTION
**Pull request by mohanelamurugan**
### Description

[MCLOUD-5818](https://jira.corp.magento.com/browse/MCLOUD-5818) :
This fixes composer running out of memory when installing multiple packages through cloud docker

### Fixed Issues (if relevant)

1. magento/magento-cloud-docker#191: Composer runs out of memory in container

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: images/php/7.1-cli/etc/php-cli.ini [hunk 1]
```diff
@@ -7,3 +7,4 @@ always_populate_raw_post_data = -1
 max_input_vars = 10000
 session.gc_probability = 1
 opcache.enable = 1
+composer_memory_limit = -1
```
Comment (shiftedreality) [inline images/php/7.1-cli/etc/php-cli.ini]: This probably should be an environment variable in `Dockerfile`

**Review by shiftedreality — CHANGES_REQUESTED** [review]

**Review by michaelcasey316 — CHANGES_REQUESTED** [review]
I agree with @shiftedreality. Having this adjustable in docker-compose.yml or the Dockerfile would be ideal.

**Comment by aepod** [comment]
I suppose it is fair to say that if we are setting a default in the
containers, we should expose it in the Dockerfile so that it can be used
from docker commands easily.

If we had a styleguide, this would be a good rule it seems. Thanks for the
feedback @michaelcasey316 and @shiftedreality
<https://github.com/shiftedreality>.

File: images/php/7.1-cli/etc/php-cli.ini (hunk 1 above)
Comment (mohanelamurugan) [reply images/php/7.1-cli/etc/php-cli.ini]: I have added an environment variable in the `Dockerfile` and used it in ini now.

**Comment by mohanelamurugan** [comment]
@shiftedreality Requested changes are implemented.

File: images/php/cli/etc/php-cli.ini [hunk 2]
```diff
@@ -7,3 +7,4 @@ always_populate_raw_post_data = -1
 max_input_vars = 10000
 session.gc_probability = 1
 opcache.enable = 1
+composer_memory_limit = COMPOSER_MEMORY_LIMIT
```
Comment (shiftedreality) [inline images/php/cli/etc/php-cli.ini]: This is probably not valid syntax, is there any sample of this notation?

Comment (shiftedreality) [inline images/php/cli/etc/php-cli.ini]: This is probably not valid syntax, is there any sample of this notation? I'm also not sure of `composer_memory_limit` exists

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: images/php/cli/etc/php-cli.ini (hunk 2 above)
Comment (mohanelamurugan) [reply images/php/cli/etc/php-cli.ini]: @shiftedreality  Here in https://github.com/magento/magento-cloud-docker/blob/develop/images/php/fpm/etc/php-fpm.ini , it have been used as `memory_limit = !PHP_MEMORY_LIMIT! ; Variable: PHP_MEMORY_LIMIT ` . I did similar implementation.

Comment (mohanelamurugan) [reply images/php/cli/etc/php-cli.ini]: @shiftedreality  Here in https://github.com/magento/magento-cloud-docker/blob/develop/images/php/fpm/etc/php-fpm.ini , it have been used as `memory_limit = !PHP_MEMORY_LIMIT! ; Variable: PHP_MEMORY_LIMIT ` . I did similar implementation.

Comment (shiftedreality) [reply images/php/cli/etc/php-cli.ini]: Yes, but `memory_limit` is supported in `php.ini`, which is not a case with `composer_memory_limit`

Please revert all `php.ini` changes. ENV variable should be enough according to https://getcomposer.org/doc/articles/troubleshooting.md

Comment (mohanelamurugan) [reply images/php/cli/etc/php-cli.ini]: okay, Sure. Its clear now. I will leave the environment variable and remove the ini changes

Comment (mohanelamurugan) [reply images/php/cli/etc/php-cli.ini]: I have fixed it now. Can you please review it

**Comment by mohanelamurugan** [comment]
@shiftedreality Now only the composer memory limit is set as environment variable without any ini file changes.

**Review by michaelcasey316 — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
@shiftedreality Maybe make sense to merge this PR first https://github.com/magento/magento-cloud-docker/pull/198 and update current PR? Otherwise we should  create task task to update COMPOSER_MEMORY_LIMIT -1 for php 7.4 or fix memory limit  in PR with php 7.4 image.

**Comment by andriyShevtsov** [comment]
Approved

**Pull request merged**